### PR TITLE
balloon_check: Fix bug when utils_misc.wait_for() timeout

### PR DESCRIPTION
**Pull request by huangyum**
ID: 1433187
Signed-off-by: Yumei Huang <yuhuang@redhat.com>

### CODE REVIEW

File: qemu/tests/balloon_check.py [hunk 1]
```diff
@@ -17,6 +17,7 @@ class BallooningTest(MemoryBaseTest):
 
     def __init__(self, test, params, env):
         self.test_round = 0
+        self.test = test
```
Comment (luckyh) [inline qemu/tests/balloon_check.py]: I guess the superclass would initialize `self.test` so you don't have to perform it again.

Comment (huangyum) [reply qemu/tests/balloon_check.py]: Yes, you are right. Updated!

**Comment by luckyh** [comment]
@huangyum can you please provide the test result? thanks.

**Comment by huangyum** [comment]
Test with win2012r2 guest, when utils_misc.wait_for() timeout, case failed as expected:

02:27:57 ERROR| FAIL 01-smp_16.4096m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_blk.up.virtio_net.Win2012.x86_64.r2.io-github-autotest-qemu.balloon_check.balloon_evict.balloon-reboot -> TestFail: After sub test, memory check failed

**Pull request merged**